### ISSUE TITLE
Mac build tests

### ISSUE THREAD

**Issue by BenRacicot** [maikramer/Materialize#10]
Hey @maikramer. I've opened the latest Dropbox build from @fouman75.
I believe it is completely non-functional: 

<img width="1601" alt="Screen Shot 2019-08-31 at 8 05 42 AM" src="https://user-images.githubusercontent.com/1815385/64063754-a55fe880-cbc6-11e9-85a1-9c64a3a6a44f.png">

The above is how it looks when opening the app, then the red text is from: adding a square png of rocks as a height map image -> clicking save

Also, why does the software ask to accept incoming network requests?
<img width="424" alt="Screen Shot 2019-08-31 at 8 11 26 AM" src="https://user-images.githubusercontent.com/1815385/64063775-f4a61900-cbc6-11e9-99bf-78f3e065f573.png">

**Comment by fouman75** [comment]
The log file should be produced when running the app. Next to the red message, there is a "Open Log File" button. Can you locate the file from there and attach it here?

**Comment by BenRacicot** [comment]
@fouman75 thank you for explaining, I hadn't noticed that. I've attached the log file.
[Player.log](https://github.com/maikramer/Materialize/files/3562656/Player.log)

I'm seeing two things
`16: Plugins: Couldn't open libc, error: dlopen(libc, 2): image not found`
May be causing
`43: NullReferenceException: Object reference not set to an instance of an object`

**Comment by fouman75** [comment]
- For network access, I believe it might be because it's a development build and you can attach a debugger to the running application. Did you notice that message when running the previous version?
- For "Couldn't open libc ...", I have the same message in my logs. We should disregard this for now as I don't think it's relevant.
- I looked at the code a bit and I see a problem with the code that saves textures. I have the same error on my side.

But before anything else, I'm very confused with your screenshot. This is not how the app should look at all.

This is what it should look like:
![Screen Shot 2019-08-31 at 3 55 12 PM](https://user-images.githubusercontent.com/18633196/64068649-f7296280-cc08-11e9-9d63-a3232bb7958a.png)

**Can you confirm if the app looks like this on your side?** There might be something really wrong if it's all black like your screenshot, regardless of the small problem with saving.

I know the app is not the friendliest to work with. Usually you would do the following:
- Open (O) a texture in the **DIFFUSE** box.
- Use the "Create" button for Height map / Click "Set as ..." when done editing.
- Use the "Create" button on any other maps ...

At any one point, click "Show Full Material" to see the results on the cube/sphere/...

Please confirm if you can at least get the basic workflow going of seeing the material in 3D.

**Comment by BenRacicot** [comment]
Yes always black, I do not see the blued image and 3D movable pane. I had watched a YT tutorial on Materialize and have been trying his workflow. Nothing works.

Here is the result of your outlined flow:
<img width="1590" alt="Screen Shot 2019-09-02 at 8 39 17 AM" src="https://user-images.githubusercontent.com/1815385/64115336-55fff080-cd5d-11e9-9eec-c63b89839644.png">

When I hit save on any process I get the OS file save prompt but no file exists when saving:
<img width="1593" alt="Screen Shot 2019-09-02 at 8 41 34 AM" src="https://user-images.githubusercontent.com/1815385/64115402-88115280-cd5d-11e9-887d-3b0e59806877.png">

No error logs on these flows.

**Comment by fouman75** [comment]
I don't have a MBP and I've never encountered the black screen problem (using iMac). I know another user reported the same problem a while back but there was no follow up. Since I don't have a MBP, there is only so much I can do ...

I have updated the build in dropbox for now. I changed the following:

- Fixed the saving bug
- I switched the app to fullscreen mode and activated a bunch of graphics/resolution options. It might be a resolution issue on MBP. That's the only thing I can think of right now. You should get a new window when starting the app to select your resolution. Try with default settings first and if it doesn't work, try different resolutions or switching from fullscreen (command-f) and resizing the window.
- Note that log file will be in ~/Library/Logs/Bounding Box Software/Materialize (with spaces).

Fingers crossed!

**Comment by maikramer** [comment]
Can you open a pull request? I would like to test in the other platforms
also just to be sure, thanks my friend, you are being great taking the app
further.
It used to be my training for my university, then for sure there is room
for make it better.

The Master branch I dont make big changes, it's for use without big
headaches.

The Hdrp branch is bleeding edge and is only recomended for people wanting
to see the power of unity hdrp.

I don't document too much, but any doubt about the code I will be happy to
help.

Em seg, 2 de set de 2019 11:54, Frederick Ouimet <notifications@github.com>
escreveu:

**Comment by fouman75** [comment]
@maikramer I can't seem to push a branch to your repo. 

> ERROR: Permission to maikramer/Materialize.git denied to fouman75.

Anything you can check on your side? I've been checking my ssh keys and everything seems in order. Let me know!

**Comment by maikramer** [comment]
You need to open a pull request, you are trying to push the commit
directly. Look for a tutorial on Internet.

Em qui, 5 de set de 2019 12:22, Frederick Ouimet <notifications@github.com>
escreveu:

**Comment by BenRacicot** [comment]
@fouman75 Saw this article on how to contribute to open source on [css-tricks](https://css-tricks.com/how-to-contribute-to-an-open-source-project/). It has guidance on the PR setup. I hope you are willing to keep helping this project, it needs your skills.

**Comment by fouman75** [comment]
@BenRacicot thanks for the link, I figured it out.

Did you get a chance to test the version I posted 2 weeks ago? Just curious if it's working 😁

**Comment by BenRacicot** [comment]
Hey @fouman75 thank you for asking. I thought I had until you asked. This new version did give me options such as resolution and windowed mode upon startup. However, the resolution is not scaled appropriately:

<img width="1682" alt="Screen Shot 2019-09-18 at 6 08 54 PM" src="https://user-images.githubusercontent.com/1815385/65189657-95a62800-da3f-11e9-967c-4816cb3d2053.png">

This is my chosen windowed -> 1440 resolution. The rocks image was uploaded in the first pane and saved. The outputed, saved image is what is shown.

I do not have a ~/Library/Logs/Bounding Box Software/ folder.

**Comment by fouman75** [comment]
@BenRacicot  Did you try other settings? Notably without windowed mode (unchecked) and resolution of 1920x1080 if available (or any 16:9)? Without even loading any images, if it's all black, it's not working. Right from launching the app, you should see the unity logo, then a bounding box splash screen and then you should see the UI with a 3D box and gradient background.

As for the log folder, I can't say. I just ran the app from scratch and I have the folder on my system:
From Terminal:
cd ~/Library/Logs/
cd Bounding\ Box\ Software/

Or in Finder under Users/<Username>/Library/Logs ...

Let me know! :)

**Comment by Fastfuel** [comment]
Hi, 
I dowloaded and tested @fouman75 's dropbox version and had the same problem as @BenRacicot  : black background and not usable (but not crashing) on my rig : 10.14.6 / RX580. 

Disabling the "Enable Post Processing by default" option in the settings (down right button) fixed the problem and displayed the material preview and the background at startup. I then successfully loaded a diffuse texture and created the height and normal maps before re-enabling the post-processing in the Post Process button (up lefty).

An important point : Unity is trying to phone cdp.cloud.unity3d.com port 443 for usage and stats without any warning. [Thats raised some concerns about Privacy](https://forum.unity.com/threads/unable-to-resolve-host-cdp-cloud-unity3d-com.498365/#post-5243300). 

Thanks @fouman75 for your work.

**Comment by fouman75** [comment]
@Fastfuel This is great, thanks for finding the issue! I made a new build where PostProcessing is now disabled by default.

I have also removed a setting to "hopefully" prevent Unity's data gathering.

The built is on [Dropbox](https://www.dropbox.com/s/4buvficjn2tfxob/materialize_0.3.7.app.zip?dl=0)

@maikramer: 

- I think we should close this issue. If new bugs occur, they should go in separate threads.
- Will you be hosting the OSX file on this repo as well?
- I didn't want to merge the OSX settings I changed into the main branch. Should we have a different branch for OSX? Not sure it would really hurt other platforms but don't really want to disable automatic PostProcessing for all because of a MacBook Pro issue.

**Comment by Fastfuel** [comment]
@fouman75 
> This is great, thanks for finding the issue! I made a new build where PostProcessing is now disabled by default.

Working for me, but the Materialize Configuration screen (with resolutions) is not displayed any more, even with option key pressed at startup :/

> I have also removed a setting to "hopefully" prevent Unity's data gathering.

I don't see any more call in the log file. Thanks !

**Comment by fouman75** [comment]
@Fastfuel For the Config screen, that feature has been deprecated by Unity. I turned it on to see if the black problem was tied to the resolution. I turned it back off in the latest build. I can activate it again if needed. In the current build, you can use Command-F to get out of full screen and resize the window if needed.

**Comment by maikramer** [comment]
Hi, i`m merging the pull request and i will add some new features also.
@fourman I will try to add the Unity Post Processing stack as a option, will try to add it today.

**Comment by maikramer** [comment]
> @Fastfuel This is great, thanks for finding the issue! I made a new build where PostProcessing is now disabled by default.
> 
> I have also removed a setting to "hopefully" prevent Unity's data gathering.
> 
> The built is on [Dropbox](https://www.dropbox.com/s/4buvficjn2tfxob/materialize_0.3.7.app.zip?dl=0)
> 
> @maikramer:
> 
>     * I think we should close this issue. If new bugs occur, they should go in separate threads.
> 
>     * Will you be hosting the OSX file on this repo as well?
> 
>     * I didn't want to merge the OSX settings I changed into the main branch. Should we have a different branch for OSX? Not sure it would really hurt other platforms but don't really want to disable automatic PostProcessing for all because of a MacBook Pro issue.

Can you open a pull request with the options you changed? then i can check and see if i will merge.
